### ISSUE TITLE
MultiLayer Graph Attention Networks

### ISSUE THREAD

**Issue by adrinta** [danielegrattarola/spektral#92]
Hi @danielegrattarola,

I am wondering, how can i implement Multi-Layer GraphAttention, i try to implement it but i doubt if my code is already right or not?

- I am confused while i pass the feature matrix **X_0**, should i pass another adjacency matrix?
- What is the purpose of the adjacency matrix while it is replaced by attention CMIIW?

Thank you

My Multilayer GraphAttention 
```
X_in =  Input(shape=(10,100))
A_in = Input(shape=(10,10))
X_0 = GraphAttention(16, attn_heads=4 )([X_in, A_in])
X_0 = GraphAttention(n_classes, attn_heads=4, concat_heads=False, activation='softmax')([X_0, A_in])

model = Model(inputs=[X_in, A_in], outputs=X_0)
```

**Comment by adrinta** [comment]
I have another question

**Model1:**

```
X_in =  Input(shape=(10,100))
A_in = Input(shape=(10,10))
X_0 = GraphAttention(16, attn_heads=4 )([X_in, A_in])
X_0 = GraphAttention(n_classes, attn_heads=4, concat_heads=False, activation='softmax')([X_0, A_in])

model = Model(inputs=[X_in, A_in], outputs=X_0)

model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              weighted_metrics=['acc'])
model.summary()

model.fit([X, A], y,
          sample_weight=train_mask,
          validation_data=validation_data,
          shuffle=False)
```

**Model2**

```
X_in =  Input(shape=(100,))
A_in = Input(shape=(10,))
X_0 = GraphAttention(16, attn_heads=4 )([X_in, A_in])
X_0 = GraphAttention(n_classes, attn_heads=4, concat_heads=False, activation='softmax')([X_0, A_in])

model = Model(inputs=[X_in, A_in], outputs=X_0)

model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              weighted_metrics=['acc'])
model.summary()

model.fit([X, A], y,
          sample_weight=train_mask,
          validation_data=validation_data,
          batch_size=**10**,
          shuffle=False)
```

what is difference Model1 and Model2, do we have to always use batch size to declare number of matrix row?
I have tried both model and get different accuracy score, Model2 is better by significance difference.

Can I get same result using Model1 with some adjustment?
i ask this because i want to produce this architecture [https://ibb.co/zGHpysc](url), and i have problem with batch_size if i use Model2 combining the model.

The architecture is from this paper [https://openaccess.thecvf.com/content_CVPR_2019/papers/Chen_Multi-Label_Image_Recognition_With_Graph_Convolutional_Networks_CVPR_2019_paper.pdf](url)

Is possible to produce the architecture using Spektral as part of it?

Thank you very much
